bindings/go/evmc/evmc.go

```diff
@@ -195,29 +195,48 @@ func (vm *VM) SetOption(name string, value string) (err error) {
 	return err
 }
 
-func (vm *VM) Execute(ctx HostContext, rev Revision,
-	kind CallKind, static bool, depth int, gas int64,
-	recipient Address, sender Address, input []byte, value Hash,
-	code []byte) (output []byte, gasLeft int64, err error) {
+type Parameters struct {
+	Context   HostContext
+	Revision  Revision
+	Kind      CallKind
+	Static    bool
+	Depth     int
+	Gas       int64
+	Recipient Address
+	Sender    Address
+	Input     []byte
+	Value     Hash
+	Code      []byte
+}
+
+type Result struct {
+	Output    []byte
+	GasLeft   int64
+	GasRefund int64
+}
+
+func (vm *VM) Execute(params Parameters) (res Result, err error) {
```

Comment:
Although it may be more readable to use additional `Parameters` type, I'd restrain from adding it in this PR. Binding for EVMC are not so much about having beautiful Go code.

Reply:
The main motivation for this change was less the readability but to make the interface future-proof for further modifications. Using a struct as a parameter list would enable additional (optional) parameters to be added in the future without breaking compatibility with existing code using the function -- analog to the argument why the result type is a struct.

Reply:
Ok, I get it. Although I'm not sure this is good idea. There is a chance then that a modified API will compile for users but behave differently or incorrectly. Wouldn't you prefer a compiler error so that you can check what has changed in EVMC?

I'm not saying no to this, but it should go to separate PR anyway :)

Reply:
It would only work for optional parameters with sensible defaults -- things like execution hints. Maybe only those should be grouped into a parameter struct.

For now, I removed the parameter struct again.